controllers/scripts/common-env.sh

```diff
@@ -103,8 +103,60 @@ export MAPPED_PORT="$POD_PORT"
 export MAPPED_TLSPORT="$POD_TLSPORT"
 {{- end}}
 
+parseCustomNetworkIP() {
+  local networks=$1
+  networks="$(echo "${networks}" | sed 's/\[//g'| sed 's/\]//g')"
+
+  DATA="$(curl --cacert $CA_CERT -H "Authorization: Bearer $TOKEN" "$KUBE_API_SERVER/api/v1/namespaces/$NAMESPACE/pods/$MY_POD_NAME")"
+
+  NETWORKIPS="$(echo $DATA | python3 -c "import sys, json
+data = json.load(sys.stdin);
+networks = '${networks}';
+networks = networks.strip().split(' ')
+def getNetworkIP(data, networks):
+    annotations = data['metadata']['annotations']
+    key = 'k8s.v1.cni.cncf.io/network-status'
+    if key in annotations:
+      netIPs = []
+      for net in json.loads(annotations[key]):
+        if net['name'] in networks:
+          netIPs.extend(net['ips'])
+      return ' '.join(netIPs)
```

Comment:
Referred network is already checked in webhook to fail fast.
If CNI is not working as expected and netIPs is empty then empty value will be updated in corresponding interface in aerospike.conf and server will crash

Reply:
let's crash in init itself. That will be easier to debug the problem. Also if the user has given a wrong network name and a right network name in the networkNames and podAnnotations. Then netIPs will have the ips from the right network only and will be added to the config file. The server will not crash in this case. We should fail in that case otherwise user will not know if the network name is wrong.